src/App/Fossa/API/BuildWait.hs

```diff
@@ -131,3 +159,122 @@ timeout ::
   IO a ->
   IO (Maybe a)
 timeout seconds act = either id id <$> Async.race (Just <$> act) (threadDelay (seconds * 1_000_000) $> Nothing)
+
+-- | Wait for either a normal build completion or a monorepo scan completion.
+-- Try to detect the correct method, use provided fallback
+waitForScanCompletion' ::
+  ( Has Diagnostics sig m
+  , Has (Lift IO) sig m
+  , Has Logger sig m
+  , Has StickyLogger sig m
+  ) =>
+  ApiOpts ->
+  ProjectRevision ->
+  Cancel ->
+  m ()
+waitForScanCompletion' apiopts revision cancelFlag = do
+  -- Route is new, this may fail on on-prem if they haven't updated
+  project <- recover $ Fossa.getProject apiopts revision
+
+  -- Try inferring, fallback to standard.
+  let runAsMonorepo = maybe False Fossa.projectIsMonorepo project
+
+  if runAsMonorepo
+    then waitForMonorepoScan' apiopts revision cancelFlag
+    else waitForBuild' apiopts revision cancelFlag
+
+-- | Wait for a "normal" (non-VPS) build completion
+waitForBuild' ::
+  ( Has Diagnostics sig m
+  , Has (Lift IO) sig m
+  , Has Logger sig m
+  , Has StickyLogger sig m
+  ) =>
+  ApiOpts ->
+  ProjectRevision ->
+  Cancel ->
+  m ()
+waitForBuild' apiOpts revision cancelFlag = do
+  checkForTimeout cancelFlag
+  build <- Fossa.getLatestBuild apiOpts revision
+
+  case Fossa.buildTaskStatus (Fossa.buildTask build) of
+    Fossa.StatusSucceeded -> pure ()
+    Fossa.StatusFailed -> fatal BuildFailed
+    otherStatus -> do
+      logSticky' $ "[ Waiting for build completion... last status: " <> viaShow otherStatus <> " ]"
+      sendIO $ threadDelay (pollDelaySeconds * 1_000_000)
+      waitForBuild' apiOpts revision cancelFlag
+
+-- | Wait for monorepo scan completion
+waitForMonorepoScan' ::
+  ( Has Diagnostics sig m
+  , Has (Lift IO) sig m
+  , Has Logger sig m
+  , Has StickyLogger sig m
+  ) =>
+  ApiOpts ->
+  ProjectRevision ->
+  Cancel ->
+  m ()
+waitForMonorepoScan' apiOpts revision cancelFlag = do
+  checkForTimeout cancelFlag
+  Fossa.Organization orgId _ <- Fossa.getOrganization apiOpts
+  let locator = VPSCore.createLocator (projectName revision) orgId
+
+  logSticky' "[ Getting latest scan ID ]"
+  scan <- ScotlandYard.getLatestScan apiOpts locator (projectRevision revision)
+
+  logSticky' "[ Waiting for monorepo scan... ]"
+  waitForSherlockScan apiOpts locator (ScotlandYard.responseScanId scan)
+
+waitForIssues' ::
```

Comment:
Leaving a note that this should be changed `waitForIssues` before you merge and remove the other `waitForIssues`

Reply:
Already covered in the following PR.  See [this commit](https://github.com/fossas/fossa-cli/pull/754/commits/573860d9c41652955a85ce16a007442131f2c640).

Reply:
I wish I had done it in this PR, but I had already based too much work on it after I did it.  Also, I couldn't remove these too early, since other places in the codebase depend on these, and that would prevent this PR from building successfully.